### PR TITLE
DATAACCESS : Writing An Repositories To The Post Stars

### LINKED ISSUE #182: DATAACCESS : Writing An Repositories To The Post Stars

### PR DESCRIPTION
**Pull request by Bakhtiyorovna**
Closes #182

### CODE REVIEW

File: src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+namespace GreenSale.DataAccess.Repositories.BuyerPosts;
+
+public class BuyerPostStarRepository : BaseRepository, IBuyerPostStarRepository
+{
+    public async Task<long> CountAsync()
+    {
+        try
+        {
+            await _connection.OpenAsync();
```
Comment (rustambekovich) [inline src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs:18]: empty qoyish kerak

**Review by ShohruhUzDev — APPROVED** [review]

File: src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        {
+            await _connection.CloseAsync();
+        }
+    }
+
+    public async Task<long> GetAllPostIdCountAsync(long id)
+    {
+        try
+        {
+            await _connection.OpenAsync();
```
Comment (rustambekovich) [inline src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs:106]: bu yerda ham

File: src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            await _connection.CloseAsync();
+        }
+    }
+
+    public async Task<BuyerPostStars> GetByIdAsync(long Id)
+    {
+        try
+        {
+            await _connection.OpenAsync();
+            string query = $"SELECT * FROM public.buyerpoststars where id=@ID;";
```
Comment (rustambekovich) [inline src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs:127]: bu yer ham

File: src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        {
+            await _connection.CloseAsync();
+        }
+    }
+
+    public async Task<long> GetIdAsync(long userid, long postid)
+    {
+        try
+        {
+            await _connection.OpenAsync();
```
Comment (rustambekovich) [inline src/GreenSale.DataAccess/Repositories/BuyerPosts/BuyerPostStarRepository.cs:146]: this is

**Pull request merged**